lib/instrumentation/modules/cassandra-driver.js

```diff
@@ -13,14 +13,31 @@ module.exports = function (cassandra, agent, { version, enabled }) {
   }
 
   if (cassandra.Client) {
-    shimmer.wrap(cassandra.Client.prototype, 'connect', wrapConnect)
+    if (semver.gte(version, '4.4.0')) {
+      // Prior to v4.4.0, the regular `connect` function would be called by the
+      // other functions (e.g. `execute`). In newer versions an internal
+      // `_connect` function is called instead (this is also called by
+      // `connect`).
+      shimmer.wrap(cassandra.Client.prototype, '_connect', wrapAsyncConnect)
+    } else {
+      shimmer.wrap(cassandra.Client.prototype, 'connect', wrapConnect)
+    }
     shimmer.wrap(cassandra.Client.prototype, 'execute', wrapExecute)
     shimmer.wrap(cassandra.Client.prototype, 'eachRow', wrapEachRow)
     shimmer.wrap(cassandra.Client.prototype, 'batch', wrapBatch)
   }
 
   return cassandra
 
+  function wrapAsyncConnect (original) {
+    return async function wrappedAsyncConnect () {
+      const span = agent.startSpan('Cassandra: Connect', 'db', 'cassandra', 'connect')
+      const ret = await original.apply(this, arguments)
```

Comment:
just a question, don't we need to explicitly add try catch here and end span?

Reply:
Fixed